### PR TITLE
LRQA-76381 | master

### PR DESCRIPTION
**Pull request by lilylcl**
https://issues.liferay.com/browse/LRQA-76381
Hi @patriciajeanperez , the config file of Site Configuration takes longer than Instance Configuration to take effect (about 3 to 5 seconds), I did not find a good wait function to hold it, so I had to pause it for 5 seconds before refresh.

### CODE REVIEW

**Comment by lilylcl** [comment]
ci:test:sf

**Comment by patriciajeanperez** [comment]
Just started reviewing :)

File: portal-web/test/functional/com/liferay/portalweb/macros/OSGiConfig.macro [hunk 1]
```diff
@@ -32,4 +32,42 @@ ${configList}''';
...
+		var filePath = FileUtil.getCanonicalPath(${tempDir});
+
+		var filePath = "${filePath}/${osgiConfigFileName}";
+
+		Variables.assertDefined(parameterList = "${tempDir},${osgiConfigFileName}");
+
+		FileUtil.replaceStringInFile("${filePath}", "${oldValue}", "${newValue}");
+	}
+
+	macro viewNoOSGiConfig {
```
Comment (patriciajeanperez) [inline portal-web/test/functional/com/liferay/portalweb/macros/OSGiConfig.macro:47]: Can we rename this to something more descriptive? This one sounds like we are asserting there is no config file. Perhaps something like assertTextNotInOSGIConfig and assertTextInOSGIConfig?

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/portalconfiguration/systemmanagement/InstanceSettings.testcase [hunk 2]
```diff
@@ -535,6 +535,86 @@ definition {
...
+				category = "System",
+				panel = "Control Panel",
+				portlet = "Virtual Instances");
+
+			PortalInstances.addCP(
+				mailDomain = "www.able.com",
+				virtualHost = "www.able.com",
+				webId = "www.able.com");
+
+			PortalInstances.viewCP(virtualHost = "www.able.com");
```
Comment (patriciajeanperez) [inline portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/portalconfiguration/systemmanagement/InstanceSettings.testcase:573]: @lilylcl I think we can skip these assertions. Making sure the portal instance is there with the appropriate details is already covered in Portal Instances tests.

**Review by patriciajeanperez — COMMENTED** [review]
@lilylcl Another set of clearly written and clean test cases :) Good job, Lily! I just had a few feedback and suggestions. Please review them and let me know what you think.

@sharryxue Since these are test cases you wrote, do you mind taking another look and see if there is anything I missed? Thank you!

File: portal-web/test/functional/com/liferay/portalweb/macros/OSGiConfig.macro [hunk 3]
```diff
@@ -32,4 +32,42 @@ ${configList}''';
...
 	}
 
+	macro editOSGiConfigFileFromTempFolder {
+		var tempDir = PropsUtil.get("output.dir.name");
+
+		var filePath = FileUtil.getCanonicalPath(${tempDir});
+
+		var filePath = "${filePath}/${osgiConfigFileName}";
+
+		Variables.assertDefined(parameterList = "${tempDir},${osgiConfigFileName}");
```
[the remark this answers is not in the document]
Comment (patriciajeanperez) [reply portal-web/test/functional/com/liferay/portalweb/macros/OSGiConfig.macro:42]: sorry I meant the `Variables.assertDefined`

**Review by Sharryxue — COMMENTED** [review]
Added the comments for review. Good job in writing the tests @lilylcl, they are clean and tidy.
/cc @patriciajeanperez  @lilylcl

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/portalconfiguration/systemmanagement/InstanceSettings.testcase [hunk 4]
```diff
@@ -535,6 +535,86 @@ definition {
...
+
+		task ("Given change the Instance OSGi configurations to export") {
+			PortalSettings.openInstanceSettingsAdmin();
+
+			SystemSettings.gotoConfiguration(
+				configurationCategory = "Web Content",
+				configurationName = "Web Content",
+				configurationScope = "Virtual Instance Scope");
+
+			AssertElementPresent(
```
[the remark this answers is not in the document]
Comment (lilylcl) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/portalconfiguration/systemmanagement/InstanceSettings.testcase:551]: It works! Thanks Sharry！

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/controlmenu/SystemSettings.path [hunk 5]
```diff
@@ -15,6 +15,16 @@
 	<td>//tr[@data-qa-id="row"]//strong[contains(.,'${key_configurationEntry}')]</td>
 	<td></td>
 </tr>
+<tr>
+	<td>CHARACTERS_BLACKLIST_VALUE</td>
+	<td>//textarea[contains(@id,'charactersblacklist') and contains(.,"${key_value}")]</td>
+	<td></td>
```
[the remark this answers is not in the document]
Comment (lilylcl) [reply portal-web/test/functional/com/liferay/portalweb/paths/portlet/controlmenu/SystemSettings.path:21]: The assert about charactersblacklist will be replaced by the SystemSettings.viewTextareaValue macro.

File: portal-web/test/functional/com/liferay/portalweb/macros/OSGiConfig.macro (hunk 3 above)
[the remark this answers is not in the document]
Comment (lilylcl) [reply portal-web/test/functional/com/liferay/portalweb/macros/OSGiConfig.macro:42]: Done.

File: portal-web/test/functional/com/liferay/portalweb/macros/OSGiConfig.macro (hunk 1 above)
Comment (lilylcl) [reply portal-web/test/functional/com/liferay/portalweb/macros/OSGiConfig.macro:47]: Done.

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/portalconfiguration/systemmanagement/SystemSettings.testcase [hunk 6]
```diff
@@ -191,6 +191,170 @@ definition {
...
+				configurationScope = "Site Scope");
+
+			Click(locator1 = "Button#REMOVE_BUTTON");
+
+			Click(locator1 = "Button#SAVE");
+
+			LexiconEntry.gotoEllipsisMenuItem(menuItem = "Export");
+		}
+
+		task ("And a new Virtual Instance is created") {
```
[the remark this answers is not in the document]
Comment (lilylcl) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/portalconfiguration/systemmanagement/SystemSettings.testcase:214]: Done.

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/portalconfiguration/systemmanagement/InstanceSettings.testcase (hunk 2 above)
Comment (lilylcl) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/portalconfiguration/systemmanagement/InstanceSettings.testcase:573]: I think it's OK, skip all view instance assertion.

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/portalconfiguration/systemmanagement/SystemSettings.testcase [hunk 7]
```diff
@@ -191,6 +191,76 @@ definition {
...
+				oldValue = "liferay.com--Guest",
+				osgiConfigFileName = "${exportConfigFileName}");
+
+			OSGiConfig.copyOSGiConfigFile(
+				osgiConfigFileBaseDir = "${exportFileDirectory}",
+				osgiConfigFileName = "${exportConfigFileName}");
+		}
+
+		task ("And login to the new instance and go to Accounts in the site settings") {
+			Navigator.openSpecificURL(url = "http://localhost:8080/group/site-name/~/control_panel/manage/-/site/settings");
```
[the remark this answers is not in the document]
Comment (lilylcl) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/portalconfiguration/systemmanagement/SystemSettings.testcase:233]: Got it & done.

File: portal-web/test/functional/com/liferay/portalweb/macros/OSGiConfig.macro [hunk 8]
```diff
@@ -32,4 +32,42 @@ ${configList}''';
...
+
+		if (contains("${fileContent}", "${text}")) {
+			fail("File ${osgiConfigFileName} contains text ${text}");
+		}
+		else {
+			echo("File ${osgiConfigFileName} does not contain text ${text}");
+		}
+	}
+
+	macro viewOSGiConfig {
```
[the remark this answers is not in the document]
Comment (lilylcl) [reply portal-web/test/functional/com/liferay/portalweb/macros/OSGiConfig.macro:60]: Done.

File: portal-web/test/functional/com/liferay/portalweb/paths/portlet/controlmenu/SystemSettings.path [hunk 9]
```diff
@@ -15,6 +15,16 @@
...
+</tr>
+<tr>
+	<td>CHECK_INTERVAL_SEGMENTS_INPUT</td>
+	<td>//*[contains(text(),'Check Interval')]/following-sibling::input[contains(@class,'form-control')]</td>
+	<td></td>
+</tr>
```
[the remark this answers is not in the document]
Comment (lilylcl) [reply portal-web/test/functional/com/liferay/portalweb/paths/portlet/controlmenu/SystemSettings.path:27]: It's a mistake. I'll remove it.

File: portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/portalconfiguration/systemmanagement/SystemSettings.testcase [hunk 10]
```diff
@@ -261,6 +261,100 @@ definition {
...
+
+			Click(locator1 = "Button#REMOVE_BUTTON");
+
+			Click(locator1 = "Button#SAVE");
+
+			LexiconEntry.gotoEllipsisMenuItem(menuItem = "Export");
+		}
+
+		task ("And a new Virtual Instance is created and login to add the new site") {
+			ApplicationsMenu.gotoPortlet(
```
[the remark this answers is not in the document]
Comment (lilylcl) [reply portal-web/test/functional/com/liferay/portalweb/tests/coreinfrastructure/administration/portalconfiguration/systemmanagement/SystemSettings.testcase:285]: Done.

**Comment by lilylcl** [comment]
All done & resend [#673](https://github.com/patriciajeanperez/liferay-portal/pull/673), thanks @patriciajeanperez and @Sharryxue for review!

**Pull request closed without merging**